Cut 0.8.0

evan2645 opened the pull request:
This PR prepares us to cut SPIRE version 0.8.0, targeting commit
c8123525fba8943bb084e70317afe32986247424.

Included is an updated CHANGELOG, which pulls in some history that was
kept in the v0.7 branch but never copied over to master. This PR also
bumps the current version to 0.8.1.

Once approved and merged, we will cut an annotated tag with the contents
of the CHANGELOG for this release. Unlike previous releases, we will
delay the creation of a version branch until we begin to introduce
changes intended for 0.9.0.

The release date in the CHANGELOG is tentatively set to Monday, May
20th.

amartinezfayo commented on CHANGELOG.md:
```diff
@@ -1,5 +1,111 @@
+- Agent cache refactored to improve memory management and fix a leak (#863)
+- UpstreamCA "disk" will now reload cert and keys when needed (#903)
+- Introduced Nested SPIRE: server clusters can now be chained together (#890)
+- Fix a bug in AWS IID NodeResolver with instance profile lookup (888)
```
nit: missing # in PR number

evan2645 commented:
> I'm not sure if we have a convention to capitalize Agent and Server when we refer to SPIRE Agent and Server. It would help to have them consistently capitalized here IMO.

Aah yes, thank you for pointing this out. I updated the changelog to use lowercase agent and server.

> nit: missing # in PR number

Fixed!

> It would be great if the references to the PRs would be clickable links to the corresponding PR.

I don't know if we want to stink up the changelog with links as it may make it harder to read, however when made into a release note, GitHub automatically link-ifies the #'s: https://github.com/spiffe/spire/releases

amartinezfayo commented:
> I don't know if we want to stink up the changelog with links as it may make it harder to read, however when made into a release note, GitHub automatically link-ifies the #'s: https://github.com/spiffe/spire/releases

That's a fair point.
